xbmc/pvr/PVRGUIChannelPreview.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ *      Copyright (C) 2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "PVRGUIChannelPreview.h"
+
+#include "GUIInfoManager.h"
+#include "ServiceBroker.h"
+#include "settings/Settings.h"
+#include "settings/lib/SettingsManager.h"
+
+#include "pvr/PVRGUIActions.h"
+#include "pvr/PVRJobs.h"
+#include "pvr/PVRManager.h"
+#include "pvr/channels/PVRChannelGroup.h"
+
+namespace PVR
+{
+  void CPVRGUIChannelPreview::NextChannel(ChannelSwitchMode eSwitchMode)
+  {
+    const CPVRChannelPtr nextChannel = GetNextOrPrevChannel(true);
+    if (nextChannel)
+      PreviewChannel(nextChannel, eSwitchMode);
+  }
+
+  void CPVRGUIChannelPreview::PreviousChannel(ChannelSwitchMode eSwitchMode)
+  {
+    const CPVRChannelPtr prevChannel = GetNextOrPrevChannel(false);
+    if (prevChannel)
+      PreviewChannel(prevChannel, eSwitchMode);
+  }
+
+  CPVRChannelPtr CPVRGUIChannelPreview::GetNextOrPrevChannel(bool bNext)
+  {
+    const bool bPlayingRadio = CServiceBroker::GetPVRManager().IsPlayingRadio();
+    const bool bPlayingTV = CServiceBroker::GetPVRManager().IsPlayingTV();
+
+    if (bPlayingTV || bPlayingRadio)
+    {
+      const CPVRChannelGroupPtr group = CServiceBroker::GetPVRManager().GetPlayingGroup(bPlayingRadio);
+      if (group)
+      {
+        CSingleLock lock(m_critSection);
+        const CFileItemPtr item = bNext
+          ? group->GetByChannelUp(m_currentChannel)
+          : group->GetByChannelDown(m_currentChannel);;
+        if (item)
+          return item->GetPVRChannelInfoTag();
+      }
+    }
+    return CPVRChannelPtr();
+  }
+
+  void CPVRGUIChannelPreview::PreviewChannel(const CPVRChannelPtr channel, ChannelSwitchMode eSwitchMode)
+  {
+    if (!g_infoManager.GetShowInfo() && eSwitchMode == ChannelSwitchMode::NO_SWITCH)
+    {
+      // just show info for current channel on first info activation.
```

Comment:
we should adjust the comment, please! ATM I thought it's triggered `on first info activation` but that is not the case. We show the info for the current channel on the first up/down (`ACTION_MOVE_UP` / `ACTION_MOVE_DOWN`) action.